CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Onboarding/OnboardingViewController.swift

```diff
@@ -1,181 +1,132 @@
+        configureCollectionView()
+        configureBottomNavigation()
     }
     
-    /**
-     Convenience initializer for the `OnboardingViewController` which will set a predefined set
-     of views as the onboarding pages.
-     
-     - parameter scrollViewDelegate: The receiver for the scroll view delegate callbacks.
-     
-     - returns: A view controller instance intended to allow the user to get a brief overview over
-     the functionality provided by the Gini Capture SDK.
-     */
-    public convenience init(scrollViewDelegate: UIScrollViewDelegate?) {
-        self.init(pages: GiniConfiguration.shared.onboardingPages, scrollViewDelegate: scrollViewDelegate)
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
     }
     
-    /**
-     Returns an object initialized from data in a given unarchiver.
-     
-     - warning: Not implemented.
-     */
-    public required init?(coder aDecoder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
+    private func layoutBottomNavigationBar(_ navigationBar: UIView) {
+        navigationBar.translatesAutoresizingMaskIntoConstraints = false
+        let horizontalConstraint = navigationBar.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+        let verticalConstraint = navigationBar.topAnchor.constraint(equalTo: pageControl.bottomAnchor)
+        let widthConstraint = navigationBar.widthAnchor.constraint(equalTo: view.widthAnchor)
+        let heightConstraint = navigationBar.heightAnchor.constraint(equalToConstant: navigationBar.frame.height)
+        let bottomConstraint = navigationBar.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        NSLayoutConstraint.activate([horizontalConstraint, verticalConstraint, heightConstraint, widthConstraint, bottomConstraint])
     }
     
-    public override func loadView() {
-        super.loadView()
-        view.addSubview(scrollView)
-        scrollView.addSubview(contentView)
-        for page in self.pages {
-            contentView.addSubview(page)
+    private func configureBottomNavigation() {
+        if configuration.bottomNavigationBarEnabled {
+            removeButtons()
+            if let customBottomNavigationBar = configuration.onboardingNavigationBarBottomAdapter {
+                navigationBarBottomAdapter = customBottomNavigationBar
+            } else {
+                navigationBarBottomAdapter = DefaultOnboardingNavigationBarBottomAdapter()
+                navigationBarBottomAdapter?.setNextButtonClickedActionCallback {
+                    self.nextPage()
+                }
+                                
+                navigationBarBottomAdapter?.setSkipButtonClickedActionCallback {
+                }
+
+                if let navigationBar =
+                    navigationBarBottomAdapter?.injectedView() {
+                    view.addSubview(navigationBar)
+                    layoutBottomNavigationBar(navigationBar)
+                }
```

Comment:
These methods need to be called also for custom bottom navigation bars.

Reply:
Nice catch 👏🏻